client/ayon_core/lib/path_templates.py

```diff
@@ -460,6 +460,30 @@ def validate_value_type(value):
                 return True
         return False
 
+    @staticmethod
+    def validate_key_is_matched(key):
```

Comment:
I'm afraid we don't support inner keys in template (e.g. `{{{some[key]}}}`), so we don't need to validate it, validating that is out of scope of this PR. We also don't support `()`. So only remaining check is `[]`.

But because I like it, we might just mention it in docstring.

Reply:
Adjusted, thanks for the feedback !